### PR TITLE
Add `url_template` and `layer_options` arguments to GeoJSON display class

### PR DESCRIPTION
**Pull request by gnestor**
This would enable users to configure Leaflet and e.g. [map points on other planets](https://github.com/jupyterlab/jupyterlab_geojson/issues/27).

```py
from IPython.display import GeoJSON

GeoJSON(data={
    "type": "Feature",
    "geometry": {
        "type": "Point",
        "coordinates": [-81.327, 296.038]
    },
    "properties": {
        "name": "Inca City"
    }
}, 
url_template="http://s3-eu-west-1.amazonaws.com/whereonmars.cartodb.net/{basemap_id}/{z}/{x}/{y}.png", 
layer_options={
    "basemap_id": "celestia_mars-shaded-16k_global",
    "attribution" : "Celestia/praesepe",
    "minZoom" : 0,
    "maxZoom" : 18,
})
```

### CODE REVIEW

**Comment by gnestor** [comment]
@Carreau @takluyver This think this is ready

**Comment by Carreau** [comment]
I've added test, docs and a line in the what's new. Let me know if that's allright.

File: docs/source/whatsnew/pr/display-geojson.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+* IPython.display has gained a :any:`GeoJSON <IPython.display.GeoJSON>` object.
+  :ghpull:`10288` and :ghpull:`????`
```
Comment (takluyver) [inline docs/source/whatsnew/pr/display-geojson.rst]: #10253

File: IPython/core/tests/test_display.py [hunk 2]
```diff
@@ -28,6 +28,28 @@ def test_image_size():
...
+            }
+        },
+        url_template="http://s3-eu-west-1.amazonaws.com/whereonmars.cartodb.net/{basemap_id}/{z}/{x}/{y}.png",
+        layer_options={
+            "basemap_id": "celestia_mars-shaded-16k_global",
+            "attribution": "Celestia/praesepe",
+            "minZoom": 0,
+            "maxZoom": 18,
+        })
+    nt.assert_equal(u'<IPython.core.display.GeoJSON object>', str(g))
```
Comment (takluyver) [inline IPython/core/tests/test_display.py]: `g` -> `gj`

**Comment by Carreau** [comment]
Thanks both !

**Pull request merged**